homeassistant/components/keenetic_ndms2/router.py

```diff
@@ -0,0 +1,200 @@
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import async_track_point_in_utc_time
+import homeassistant.util.dt as dt_util
+
+from .const import (
+    CONF_CONSIDER_HOME,
+    CONF_INCLUDE_ARP,
+    CONF_INCLUDE_ASSOCIATED,
+    CONF_INTERFACES,
+    CONF_TRY_HOTSPOT,
+    DEFAULT_CONSIDER_HOME,
+    DEFAULT_INTERFACE,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class KeeneticRouter:
+    """Keenetic client Object."""
+
+    def __init__(self, hass: HomeAssistant, config_entry: ConfigEntry):
+        """Initialize the Client."""
+        self.hass = hass
+        self.config_entry = config_entry
+        self.last_devices = {}  # type: Dict[str, Device]
+        self._router_info = None  # type: Optional[RouterInfo]
+        self._client = None  # type: Optional[Client]
+        self._cancel_periodic_update = None  # type: Optional[Callable]
+        self._available = False
+        self.progress = None
+
+    @property
+    def host(self):
+        """Return the host of this hub."""
+        return self.config_entry.data[CONF_HOST]
+
+    @property
+    def device_info(self):
+        """Return the host of this hub."""
+        return {
+            "identifiers": {(DOMAIN, f"router-{self.config_entry.entry_id}")},
+            "manufacturer": self.manufacturer,
+            "model": self.model,
+            "name": self.name,
+            "sw_version": self.firmware,
+        }
+
+    @property
+    def name(self):
+        """Return the name of the hub."""
+        return self._router_info.name if self._router_info else self.host
+
+    @property
+    def model(self):
+        """Return the model of the hub."""
+        return self._router_info.model if self._router_info else None
+
+    @property
+    def firmware(self):
+        """Return the firmware of the hub."""
+        return self._router_info.fw_version if self._router_info else None
+
+    @property
+    def manufacturer(self):
+        """Return the firmware of the hub."""
+        return self._router_info.manufacturer if self._router_info else None
+
+    @property
+    def available(self):
+        """Return if the hub is connected."""
+        return self._available
+
+    @property
+    def consider_home_interval(self):
+        """Config entry option defining number of seconds from last seen to away."""
+        return timedelta(seconds=self.config_entry.options[CONF_CONSIDER_HOME])
+
+    @property
+    def signal_update(self):
+        """Event specific per router entry to signal updates."""
+        return f"keenetic-update-{self.config_entry.entry_id}"
+
+    async def async_add_defaults(self):
```

Comment:
This doesn't need to be a coroutine function since we don't await inside.
```suggestion
    @callback
    def async_add_defaults(self):
```